### PR TITLE
KRD R package

### PR DESCRIPTION
**Pull request by kdorheim**
@bpbond this not all that pretty and there is something funky going on with some of the functions, for some parameters need to be defined in the global environment which makes little sense... Also I can't figure out what the time index is for the results but that is something that would be helpful to know...

### CODE REVIEW

File: data-raw/DATASET.R [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+params$Lmin <- 0.75
+params$q <- 1
+params$StoreMinDay <- 2
+params$Smax <- 1
+params$Rfrac <- 0.2
+params$SeedlingMort <- 0.99
+params$Kleaf <- (1 / 21 / 48) / 2^2.5 ## assumes it takes 21 days to regrow at 25C
+
+default_parameters <- params
+usethis::use_data(default_parameters, overwrite = TRUE)
```
Comment (bpbond) [inline data-raw/DATASET.R:47]: OK, interesting approach.

File: tests/testthat/test-utility.R [hunk 2]
```diff
@@ -0,0 +1,9 @@
+context('utility functions')
+
+test_that("format_inputs", {
+
+  testthat::expect_error(format_inputs('fake.nc'))
```
Comment (bpbond) [inline tests/testthat/test-utility.R]: When these are run, the `testthat` namespace should be loaded I think, so don't need the `testthat::` prefix here and below?

File: vignettes/SEM_example.Rmd [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
```
Comment (bpbond) [inline vignettes/SEM_example.Rmd:1]: Oh this is awesome @kdorheim 👏

File: R/utility_fxns.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+format_inputs <- function(nc_path){
+
+  assertthat::assert_that(file.exists(nc_path))
+
+  # Import the data and format it
+  nc <- ncdf4::nc_open(nc_path)
+
+  PAR <- ncdf4::ncvar_get(nc, "PAR")
+  for (i in which(PAR < -10)) {
+    PAR[i] <- PAR[i - 1]
```
Comment (bpbond) [inline R/utility_fxns.R:11]: What about if i=1?

File: data-raw/DATASET.R [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Make a list of the default parameters from Dietz et al. paper.
```
Comment (bpbond) [inline data-raw/DATASET.R]: "Dietze"

File: R/model_functions.R [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+
+  ## evaporation (patch)
+  EVAP <- min(X[6] / timestep, 1.15 * params$gevap * (0.622 * inputs$VPD / P) / 1000) ## Bonan Eqn 13.10: rate = m/s
+  ## 1.15 kg/m3 = rho = density of air
+  ## 1000 at end converts kg/m2/s to m/s
+  ##**** CHECK UNITS (want m/s)
+  X[6] <- X[6] + inputs$precip / 1000 - EVAP * timestep
+
+  ## plant available water (trapazoidal response) (patch)
+  paw <- ifelse(X[6] > params$Wthresh, X[6] - 0.5 * params$Wthresh,
```
Comment (bpbond) [inline R/model_functions.R:39]: I know this is Mike's code, not yours, but we should replace these vector-position numbers with named constants at some point.

**Review by bpbond — APPROVED** [review]
Thanks for you great work @kdorheim

File: data-raw/DATASET.R (hunk 1 above)
Comment (kdorheim) [reply data-raw/DATASET.R:47]: I went to school in MN, by interesting do you mean bad? Cause if it is interesting bad... let's not do it. Working with a ton of parameters that have to be defined always make me nervous, there are so many places to do wrong. Personally I'd rather take something that I know works and modify it... but I also recognize that there are some negatives to doing it this way 🤷

File: R/utility_fxns.R (hunk 4 above)
Comment (kdorheim) [reply R/utility_fxns.R:11]: ummm no idea this... this is pulled directly from their code...

Comment (bpbond) [reply R/utility_fxns.R:11]: File under "not our problem" then

File: data-raw/DATASET.R (hunk 1 above)
Comment (bpbond) [reply data-raw/DATASET.R:47]: > went to school in MN

😆 
I'm not passive-aggressive'ing you. I like it.

Comment (kdorheim) [reply data-raw/DATASET.R:47]: haha okay after 4 years there whenever the work interesting is used I've gotta double check

File: data-raw/DATASET.R (hunk 5 above)
Comment (kdorheim) [reply data-raw/DATASET.R]: 🙈

**Comment by kdorheim** [comment]
Made the changes you suggested or moved comments to address latter to a github issue. Merging and closing the PR now.

**Pull request merged**